File: app/src/main/java/com/antyzero/smoksmog/ui/utils/DimenUtils.java
package com.antyzero.smoksmog.ui.utils;

import android.content.Context;
import android.content.res.Resources;
import android.support.annotation.DimenRes;

/**
 * Various tools for positioning manipulation
 */
public class DimenUtils {

    private DimenUtils() {
        throw new IllegalAccessError( "Utils class" );
    }

    public static int getNavBarHeight( Context context, @DimenRes int defaultRes ) {
        Resources resources = context.getResources();
        int resourceId = resources.getIdentifier( "navigation_bar_height", "dimen", "android" );
        return resources.getDimensionPixelSize( resourceId > 0 ? resourceId : defaultRes );
    }

    /**
     * @param context
     * @return
     */
    public static int getNavBarHeight( Context context ) {
        Resources resources = context.getResources();
        int resourceId = resources.getIdentifier( "navigation_bar_height", "dimen", "android" );
        return resourceId > 0 ? resources.getDimensionPixelSize( resourceId ) : 0;
    }

    public static int getStatusBarHeight( Context context, @DimenRes int defaultRes ) {
        Resources resources = context.getResources();
        int resourceId = resources.getIdentifier( "status_bar_height", "dimen", "android" );
        return resources.getDimensionPixelSize( resourceId > 0 ? resourceId : defaultRes );
    }

    public static int getStatusBarHeight( Context context ) {
        Resources resources = context.getResources();
        int resourceId = resources.getIdentifier( "status_bar_height", "dimen", "android" );
        return resources.getDimensionPixelSize( resourceId );
    }
}

File: app/src/main/kotlin/com/antyzero/smoksmog/ui/utils/DimenUtils.kt
package com.antyzero.smoksmog.ui.utils

import android.content.Context
import android.content.res.Resources
import android.support.annotation.DimenRes

/**
 * Various tools for positioning manipulation
 */
class DimenUtils private constructor() {

    init {
        throw IllegalAccessError("Utils class")
    }

    companion object {

        fun getNavBarHeight(context: Context, @DimenRes defaultRes: Int): Int {
            val resources = context.resources
            val resourceId = resources.getIdentifier("navigation_bar_height", "dimen", "android")
            return resources.getDimensionPixelSize(if (resourceId > 0) resourceId else defaultRes)
        }

        /**
         * @param context
         * *
         * @return
         */
        fun getNavBarHeight(context: Context): Int {
            val resources = context.resources
            val resourceId = resources.getIdentifier("navigation_bar_height", "dimen", "android")
            return if (resourceId > 0) resources.getDimensionPixelSize(resourceId) else 0
        }

        fun getStatusBarHeight(context: Context, @DimenRes defaultRes: Int): Int {
            val resources = context.resources
            val resourceId = resources.getIdentifier("status_bar_height", "dimen", "android")
            return resources.getDimensionPixelSize(if (resourceId > 0) resourceId else defaultRes)
        }

        fun getStatusBarHeight(context: Context): Int {
            val resources = context.resources
            val resourceId = resources.getIdentifier("status_bar_height", "dimen", "android")
            return resources.getDimensionPixelSize(resourceId)
        }
    }
}
